pkg/specgen/generate/container_create.go

```diff
@@ -24,11 +25,28 @@ func MakeContainer(ctx context.Context, rt *libpod.Runtime, s *specgen.SpecGener
 	// If joining a pod, retrieve the pod for use.
 	var pod *libpod.Pod
 	if s.Pod != "" {
-		foundPod, err := rt.LookupPod(s.Pod)
-		if err != nil {
-			return nil, errors.Wrapf(err, "error retrieving pod %s", s.Pod)
+		podName := s.Pod
```

Comment:
Should handle this on the frontend, not the backend, IMO - if we find the "new" sigil, we make a pod first, then make a SpecGenerator with the new pod's ID

Reply:
won't that cause all clients to implement the same logic?  Also, potentially the operation could be done atomically in the backend or at least have better cleanup on errors

Reply:
I view it as a call to the existing `pod create` API with everything defaulted except name, which shouldn't be extra code.

As for atomicity... Hmm. I don't think we actually handled that in Podmanv1, so we're not obligated to, but that does not mean we can't be better.

My main objection is that I really don't want to have to parse special sigils or other command line artifacts in Specgen - I wanted the input in Specgen to be interpreted literally, no special processing, wherever possible for simplicity.

@baude @vrothberg @rhatdan Thoughts on this one?

Reply:
I do not have strong feelings but share the general desire to keep the CLI specifics out. At some point, Podman and libpod will have separate homes and a CLI-only change should not require touching the libpod project.

Reply:
Yes I prefer this on the frontend.

Reply:
I've moved the handling of `new:` to `cmd/podman` and pushed a new version